geode-core/src/main/java/org/apache/geode/cache/ExpirationAttributes.java

```diff
@@ -91,6 +91,13 @@ public ExpirationAttributes(int expirationTime, ExpirationAction expirationActio
     }
   }
 
+  public void setTimeout(int timeout) {
```

Comment:
ExpirationAttributes is in the geode external API so you should probably not change it. It is wrong to add settors to it since the class is supposed to be immutable. The only reason its fields are not "final" was to allow serialization to be implemented. So instead of settors just use the constructor to set the timeout and action.

Reply:
setter removed